src/Uno.Toolkit.UI/Behaviors/FlipViewExtensions.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Input;
+using Microsoft.Extensions.Logging;
+using Uno.Extensions;
+using Uno.Logging;
+using Windows.System;
+using Windows.UI.ViewManagement;
+
+#if IS_WINUI
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls.Primitives;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Input;
+#else
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Input;
+#endif
+
+namespace Uno.Toolkit.UI;
+
+public static class FlipViewExtensions
+{
+	#region DependencyProperty: Next
+	public static DependencyProperty NextProperty { get; } =
+	DependencyProperty.RegisterAttached("Next", typeof(FlipView), typeof(FlipViewExtensions), new PropertyMetadata(null, OnNextChanged));
+
+	public static void SetNext(Button element, FlipView value) => element.SetValue(NextProperty, value);
+
+	public static FlipView GetNext(Button element) => (FlipView)element.GetValue(NextProperty);
+
+	#endregion
+
+	#region DependencyProperty: Previous
+	public static DependencyProperty PreviousProperty { get; } =
+	DependencyProperty.RegisterAttached("Previous", typeof(FlipView), typeof(FlipViewExtensions), new PropertyMetadata(null, OnPreviousChanged));
+
+	public static void SetPrevious(Button element, FlipView value) => element.SetValue(PreviousProperty, value);
+
+	public static FlipView GetPrevious(Button element) => (FlipView)element.GetValue(PreviousProperty);
+	#endregion
+
+	static void OnNextChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+	{
+		if (d is not ButtonBase btn || e.NewValue is null)
```

Comment:
Even if `e.NewValue == null` we'll still need to unsubscribe from the previous Click event. As long as `d` is a `ButtonBase`, we should always get past this point and at the very least we should then call `btn.Click -= OnBtnClick`. Then we can check if e.NewValue is null. If it is null then we do nothing, otherwise, re-subscribe to the Click event

Reply:
@kazo0 it will be cleaned at line `55`, so when the button is Unloaded we remove any reference to the event handler

Reply:
I may be misunderstanding something here but from what I see: right now the event registration is cleaned only once the button is unloaded, the Next/Previous properties may change values multiple times before anything is unloaded. We should only ever have one subscription to the Click event at a time. So if Next/Previous is changed, the old registration should be cleared immediately

Reply:
I see... That makes sense now, I'll address that tomorrow